### PR TITLE
Added new Pi Supply board & Fixed Ryanteck Boards

### PR DESCRIPTION
**Pull request by ryanteck**

### CODE REVIEW

**Comment by ryanteck** [comment]
This does the following:

Fix links to Motor Controller & Traffic HAT.

Adds Traffic pHAT in

File: gpiozero/boards.py [hunk 1]
```diff
@@ -1532,6 +1532,38 @@ def __init__(self, pwm=False, pin_factory=None):
...
+class TrafficpHAT(LEDBoard):
+    """
+    Extends :class:`LEDBoard` for the `Pi Supply Traffic pHAT`_: Mini traffic
+    light LEDs.
+
+    The Traffic pHAT pins are fixed and therefore there's no need to specify
+    them when constructing this class. The following example then turns on all the LEDs::
+
+        from gpiozero import TrafficpHAT
+        hat = TrafficHat()
```
Comment (bennuttall) [inline gpiozero/boards.py]: ^^^

File: gpiozero/boards.py [hunk 2]
```diff
@@ -1532,6 +1532,38 @@ def __init__(self, pwm=False, pin_factory=None):
...
+
+    :type pin_factory: Factory or None
+    :param pin_factory:
+        See :doc:`api_pins` for more information (this is an advanced feature
+        which most users can ignore).
+
+    .. _PiSupply Traffic pHAT:
+    """
+    def __init__(self, pwm=False, initial_value=False, pin_factory=None):
+        super(TrafficHat, self).__init__(
```
Comment (bennuttall) [inline gpiozero/boards.py]: ^^^

**Comment by ryanteck** [comment]
Will get these solved :+1:

File: gpiozero/boards.py [hunk 3]
```diff
@@ -1532,6 +1532,38 @@ def __init__(self, pwm=False, pin_factory=None):
...
+        If :data:`True`, construct :class:`PWMLED` instances to represent each
+        LED. If :data:`False` (the default), construct regular :class:`LED`
+        instances.
+
+    :type pin_factory: Factory or None
+    :param pin_factory:
+        See :doc:`api_pins` for more information (this is an advanced feature
+        which most users can ignore).
+
+    .. _PiSupply Traffic pHAT:
```
Comment (bennuttall) [inline gpiozero/boards.py]: missing link

File: gpiozero/boards.py [hunk 4]
```diff
@@ -1532,6 +1532,38 @@ def __init__(self, pwm=False, pin_factory=None):
         )
 
 
+class TrafficpHAT(LEDBoard):
```
Comment (bennuttall) [inline gpiozero/boards.py]: This class should extend `TrafficLights` not `LEDBoard`. Usage would just be `phat.on()` etc and `phat.green.on()` etc.

File: gpiozero/boards.py [hunk 5]
```diff
@@ -1520,7 +1520,7 @@ class TrafficHat(CompositeOutputDevice):
         See :doc:`api_pins` for more information (this is an advanced feature
         which most users can ignore).
 
-    .. _Ryanteck Traffic HAT: https://ryanteck.uk/hats/1-traffichat-0635648607122.html
+    .. _Ryanteck Traffic HAT: https://uk.pi-supply.com/products/traffic-hat-for-raspberry-pi
```
Comment (lurch) [inline gpiozero/boards.py]: I guess the `_Ryanteck Traffic HAT` here should be `_Pi Supply Traffic HAT` in order to match the earlier link?

Comment (ryanteck) [reply gpiozero/boards.py]: I'll double check but that sounds right to me. I'll get all these fixes done today.

Comment (ryanteck) [reply gpiozero/boards.py]: That naming is now fixed.

File: gpiozero/boards.py (hunk 4 above)
Comment (ryanteck) [reply gpiozero/boards.py]: Believe I've done this right now.

File: gpiozero/boards.py (hunk 3 above)
Comment (ryanteck) [reply gpiozero/boards.py]: I've added a placeholder link which will redirect when it's launched. :+1:

File: gpiozero/boards.py (hunk 2 above)
Comment (ryanteck) [reply gpiozero/boards.py]: Fixed this super

File: gpiozero/boards.py (hunk 1 above)
Comment (ryanteck) [reply gpiozero/boards.py]: And fixed as well.

**Comment by bennuttall** [comment]
TODO:

- Add basic tests
- Add to documentation

I'll add these later and we can check it works and looks right

**Comment by bennuttall** [comment]
Is there a new home for the SnowPi? The link we have is https://ryanteck.uk/raspberry-pi/114-snowpi-the-gpio-snowman-for-raspberry-pi-0635648608303.html

**Comment by ryanteck** [comment]
I'm not 100% sure with that to be honest. It was one that I originally sold to ModMyPi before moving to Pi Supply myself. 

It seems Pi Hut did stock them but say they're discontinued :(. I'll see if I can get them re-continued or such as I did like that board.

File: gpiozero/boards.py [hunk 6]
```diff
@@ -1532,6 +1532,36 @@ def __init__(self, pwm=False, pin_factory=None):
...
+
+    :type pin_factory: Factory or None
+    :param pin_factory:
+        See :doc:`api_pins` for more information (this is an advanced feature
+        which most users can ignore).
+
+    .. _PiSupply Traffic pHAT: http://pisupp.ly/trafficphat
+    """
+    def __init__(self, pwm=False, initial_value=False, pin_factory=None):
+        super(TrafficpHAT, self).__init__(25, 24, 23,
```
Comment (bennuttall) [inline gpiozero/boards.py]: Can you add kwargs to the pins? i.e. `__init__(red=25, amber=24, green=23)`

File: gpiozero/boards.py [hunk 7]
```diff
@@ -1532,6 +1532,36 @@ def __init__(self, pwm=False, pin_factory=None):
         )
 
 
+class TrafficpHAT(TrafficLights):
```
Comment (bennuttall) [inline gpiozero/boards.py]: Can you change to `pHat` so it's like `TrafficHat`?

File: gpiozero/boards.py [hunk 8]
```diff
@@ -1532,6 +1532,36 @@ def __init__(self, pwm=False, pin_factory=None):
...
+    """
+    Extends :class:`TrafficLights` for the `Pi Supply Traffic pHAT`_: Mini traffic
+    light LEDs.
+
+    The Traffic pHAT pins are fixed and therefore there's no need to specify
+    them when constructing this class. The following example then turns on all the LEDs::
+
+        from gpiozero import TrafficpHAT
+        hat = TrafficpHAT()
+        hat.lights.on()
```
Comment (bennuttall) [inline gpiozero/boards.py]: This is wrong now - should be `hat.on()` not `hat.lights.on()` and the instance should probably be called `phat` instead. May as well add an example like `phat.red.on()` too.

**Comment by bennuttall** [comment]
I'm just addressing these suggestions, will push in a mo.

**Comment by bennuttall** [comment]
Pushed to a new branch, new PR is #846

**Comment by lurch** [comment]
"**bennuttall** closed this on Mar 18, 2020"

"**codecov-commenter** commented 15 hours ago"

How bizarre!! :confounded:

**Comment by bennuttall** [comment]
Go home, codecov, you're drunk!

**Pull request closed without merging**